lib/paramedic.js

```diff
@@ -0,0 +1,232 @@
+    this.pluginsManager = new PluginsManager(this.tempFolder.name, this.storedCWD);
+    this.pluginsManager.installPlugins(this.config.getPlugins());
+    this.pluginsManager.installTestsForExistingPlugins();
+    this.pluginsManager.installSinglePlugin('cordova-plugin-test-framework');
+    this.pluginsManager.installSinglePlugin('cordova-plugin-device');
+    this.pluginsManager.installSinglePlugin(path.join(__dirname, '../paramedic-plugin'));
+};
+
+ParamedicRunner.prototype.setUpStartPage = function() {
+    logger.normal("cordova-paramedic: setting app start page to test page");
+    shell.sed('-i', 'src="index.html"', 'src="cdvtests/index.html"', 'config.xml');
+};
+
+ParamedicRunner.prototype.installPlatform = function() {
+    logger.info("cordova-paramedic: adding platform : " + this.config.getPlatform());
+    exec('cordova platform add ' + this.config.getPlatform());
+};
+
+ParamedicRunner.prototype.checkPlatformRequirements = function() {
+    logger.normal("cordova-paramedic: checking requirements for platform " + this.config.getPlatformId());
+    var result = exec('cordova requirements ' + this.config.getPlatformId());
+
+    if (result.code !== 0) 
+        throw new Error('Platform requirements check has failed!');
+};
+
+ParamedicRunner.prototype.injectReporters = function() {
+    var self = this;
+    var reporters = getReporters(self.config.getReportSavePath());
+
+    ['jasmineStarted', 'specStarted', 'specDone',
+    'suiteStarted', 'suiteDone', 'jasmineDone'].forEach(function(route) {
+        reporters.forEach(function(reporter) {
+            if (reporter[route] instanceof Function)
+                self.server.on(route, reporter[route].bind(reporter));
+        });
+    });
+};
+
+ParamedicRunner.prototype.subcribeForEvents = function() {
+    this.server.on('deviceLog', function(data) {
+        logger.verbose('device|console.' + data.type + ': '  + data.msg[0]);
+    });
+
+    this.server.on('deviceInfo', function(data) {
+        logger.normal('cordova-paramedic: Device info: ' + JSON.stringify(data));
+    });
+};
+
+ParamedicRunner.prototype.writeMedicConnectionUrl = function(url) {
+    logger.normal("cordova-paramedic: writing medic log url to project " + url);
+    fs.writeFileSync(path.join("www","medic.json"), JSON.stringify({logurl:url}));
+};
+
+ParamedicRunner.prototype.runTests = function() {
+    var self = this;
+
+    return Q.promise(function(resolve, reject) {
+        self.server.on('jasmineDone', function(data) {
+            logger.info('cordova-paramedic: tests have been completed');
+
+            var isTestPassed = data.specResults.specFailed === 0;
+
+            resolve(isTestPassed);
+        });
+
+        self.server.on('disconnect', function() {
+            reject(new Error('device is disconnected before passing the tests'));
+        });
+
+        var command = self.getCommandForStartingTests();
+        logger.normal('cordova-paramedic: running command ' + command);
```

Comment:
Propose to leave as-is - we have several methods we will need to override (info, debug, normal, ..) and also several prefixes ('cordova-paramedic', 'local-server')
PS. This is out of scope of this PR as it does not aim to change originally implemented logging approach, we can create another PR to handle this.